Create a new REUSE Specification (#23)
fsfe/reuse-docs: spec/spec.md

<!--
SPDX-Copyright: 2019 Free Software Foundation Europe e.V.

SPDX-License-Identifier: CC-BY-SA-4.0
-->

# REUSE Specification

This specification defines a standardized method for declaring copyright and
licensing for software projects. The goal of the specification is to have
unambiguous, machine-readable copyright and licensing information for each
individual file in a project.

> 

Author: I'm sorry I don't understand well enough what you mean here.

> 

Reviewer: I think @silverhook means that there is a big benefit of having copyright information **in** the files, not somewhere else like a DEP-5 file, because they will retain even if just a single file is copied from a project. Right?

> 

Reviewer 2: > I think @silverhook means that there is a big benefit of having copyright information **in** the files, not somewhere else like a DEP-5 file, because they will retain even if just a single file is copied from a project. Right?

Yes, that’s exactly what I meant with that suggestion.

## Definitions

> 

Author: LGTM

> 

Author: Added in ec3d5449d2fcd49fea6a47b95a3b85f953e1777f

These are the definitions for some of the terms used in this specification:

> - Project --- ...

Author: I don't think we can skip defining that, because it's such a central term.

"Project" is the only word I know for "the directory containing all the source files for a given program".

> - Project --- ...

Reviewer 2: Package may indeed be more accurate. Especially, if we could piggy-back on SPDX Spec here. @goneall, I can’t find where “Package” itself (as opposed to “Package Information”, “Package Name”, “Package Version”, etc.) is specified in SPDX Spec though. Am I missing something?

> - Project --- ...

Reviewer 3: I also looked for a definition in the spec and did not see a concise definition of a package.  I do recall discussing this at length during the spec development.  Something we may want to add to the spec.

> - Project --- ...

Reviewer 2: @goneall, in that case, I think it would be best if both projects use the same definition, until SPDX adopts it in a published spec. Then REUSE can simply refer to that definition.

> - License file --- ...

Author: You're right, I kind of forgot that common practice. LGTM, and will double-check this a few times after the suggestions are committed.

> - License file --- ...

Author: Added in 3614756e120551c5d59917ef6c18df1fe1eb5289

> <!-- Also put a link to the SPDX Specification here somewhere -->

Author: Agree, LGTM

> <!-- Also put a link to the SPDX Specification here somewhere -->

Author: Added in 3614756e120551c5d59917ef6c18df1fe1eb5289

> <!-- Also DEP5 -->

Reviewer 2: ```suggestion
- DEP5 File --- `.reuse/dep5` file, which apart from its filename and location, follows the definition of [Machine-readable `debian/copyright` file, Version 1.0](https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/)
```

Author: We kind of do a subset of the DEP5 spec that possibly isn't entirely compatible. We mandate SPDX expressions in the REUSE spec, and those expressions are NOT in the DEP5 spec, but are supported by the library we use (`python-debian`).

We also mandate the format of the value following the Copyright tag, but the tool currently doesn't verify this.

Reviewer 2: @carmenbianca, OK, that makes sense to me. But then it would make even more sense to also define the “REUSE DEP5” file, as the standard DEP5 with the following specifics, to make it more compatible with SPDX. If nothing else, to avoid confusion with the vanilla DEP5 files and spec.

## License files

> A project must include a license file for every license under which files in the

Author: Closing this and will implement this.

project are licensed.

> Each license file should be placed in the `LICENSES/` directory in the root of

Author: I reckon this was to meet people halfway.

If we follow this suggestion and make people use `LICENSES/[identifier].txt`, we don't really need `Valid-License-Identifier` anymore.

> Each license file should be placed in the `LICENSES/` directory in the root of

Reviewer 2: > If we follow this suggestion and make people use `LICENSES/[identifier].txt`, we don't really need `Valid-License-Identifier` anymore.

That is a huge part of the reason why I’m suggesting it :smile_cat:

If there are use cases where this makes sense, we should keep it. But I can’t find any. I hope @jonasob or someone else involved with REUSE 1.0 could provide some clarity here.

> the project. The name of the license file should be the SPDX identifier of the

Author: Closing this because another conversation is open about the same thing.

> license followed by a file extension (example: `GPL-3.0-or-later.txt`).

Author: LGTM

> license followed by a file extension (example: `GPL-3.0-or-later.txt`).

Reviewer 2: I mean, I’d love to push for a “MUST” here, but I see how we need to accommodate at least to MarkDown, ASCIIdoc etc. I would really like to keep it easily machine-readable. Perhaps add that the “if it cannot be simple plain text, at the very least it MUST be in a file format that is easily both human- and machine-readable”

> license followed by a file extension (example: `GPL-3.0-or-later.txt`).

Author: To my best understanding, markdown and asciidoc are also plain-text. It's just plain text with a certain syntax.

I can't think of many licences that aren't plain-text. I can only think of CC, which does have a plain-text version, but XHTML is the canonical version.

https://creativecommons.org/2014/01/07/plaintext-versions-of-creative-commons-4-0-licenses/

> license followed by a file extension (example: `GPL-3.0-or-later.txt`).

Author: Fixed in cf341030123f3b566765b7e69eaa90c9543ead64.

@silverhook Can you create a new issue about mandating that licenses be plain-text?

> license followed by a file extension (example: `GPL-3.0-or-later.txt`).

Reviewer 2: I opened such an issue here: https://github.com/fsfe/reuse-docs/issues/24

<!-- FIXME: Rephrase "above recommendation". Maybe use paragraph numbers? -->

A license file whose name does not conform to the above recommendation must
contain, at the start of the file, one or more `Valid-License-Identifier` tags,

> followed by a `License-Text` tag, followed by the license text. Example:

Author: Good clarification, LGTM

> followed by a `License-Text` tag, followed by the license text. Example:

Author: Fixed in 3996c8d485a438641f997164debc08467693a194

```
Valid-License-Identifier: GPL-3.0-only
Valid-License-Identifier: GPL-3.0-or-later
License-Text:

                    GNU GENERAL PUBLIC LICENSE
                       Version 3, 29 June 2007

 Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
 Everyone is permitted to copy and distribute verbatim copies
 of this license document, but changing it is not allowed.

[...]
```

> A project may contain a single license file outside of the `LICENSES/` directory

Reviewer: True, I commented on the same argument in the FAQ PR.

We can make it a MUST, and wait for comments from outside to see whether we can uphold this against the resistance I am experiencing ;)

> A project may contain a single license file outside of the `LICENSES/` directory

Reviewer 2: > We can make it a MUST, and wait for comments from outside to see whether we can uphold this against the resistance I am expecting ;)

Personally I would be in favour of that, but let’s first see if anyone complains in this PR.

> in the file `COPYING`, `COPYRIGHT`, `LICENSE` or `LICENCE` in the root of the

Reviewer: Hm, Linux for example also has a COPYING file, which summarising the situation:

https://github.com/torvalds/linux/blob/master/COPYING

> in the file `COPYING`, `COPYRIGHT`, `LICENSE` or `LICENCE` in the root of the

Reviewer 2: @mxmehl, exactly. Linux’ `COPYING` file is not “A single License File” in the sense of the “a file containing the text of a license”, as its (proposed) definition is in REUSE Spec 3.0.

The question is whether REUSE Spec should also cover how people _summarise_ the licensing situation. I think it should be outside of scope, but we could provide some suggestions in the Tutorial and FAQ.

project.

> If a license does not exist in the SPDX License List, its SPDX identifier is

Author: LGTM, though I'm not sure if we should hyperlink the SPDX License List when it's already referred to at the start of the document.

> If a license does not exist in the SPDX License List, its SPDX identifier is

Author: Fixed in 058dda3b6825eeb20b0626455928efa9c052ac12

`LicenseRef-[idstring]` as defined by the SPDX Specification.

A project must not include license files for licenses under which none of the
files in the project are licensed.

Everything that applies to licenses in this section also applies to license

> exceptions, with the following exceptions:

Reviewer 2: ```suggestion
exceptions, with the following specifics:
```

Author: Fixed in 2069aa1d0564fb448c4e37c637d9d128518f7e6b

> - Exception files must be placed in the `LICENSES/` directory in the root of the

Reviewer 2: ```suggestion
- Exception files MUST also be placed in the `LICENSES/` directory in the root of the
```

  project.

- The tags `Valid-Exception-Identifier` and `Exception-Text` must be used
  instead of `Valid-License-Identifier` and `License-Text`.

> <!-- Can we use LicenseRef for exceptions? -->

Reviewer 2: I would expect so, yes. The other option I see could be an `ExceptionRef`, which I can’t find in the SPDX spec.

@jlovejoy, @goneall, any ideas? (BTW, I would have included Kate, but can’t find her GitHub handle)

Author: @kestewart 

My interpretation of the spec limits LicenseRef to licenses only (i.e., no exceptions).

Author: Makes sense to me. It's not a huge loss, but it adds a tiny edge case.

Author: Fixed in 5b933e5e80bff6174765f31def4f380ed6ba8d8b

<!-- TODO: Change "copyright and licensing information" to something shorter?
Shorten it to CALI? -->

## Copyright and licensing information

Each file in the project must have copyright and licensing information
associated with it, except the following files:

> - The license files.

Reviewer 2: ```suggestion
- The License Files.
```

- The files belonging to the project's version control system (example:
  `.git/`).

- The files ignored by the version control system (example: `.gitignore`).

> - The files in the `.reuse/` directory in the root of the project.

Reviewer 2: Apart from the DEP5 file, what else do we allow to be stored there?

Author: In the future, a tiny configuration file, and comment header templates.

Reviewer 2: Then we should also put what goes into the `.reuse/` directory into the spec, just in case.

Author: @silverhook If I could predict the future, then yes :) I don't yet know exactly what the names of those files will be. I think this catch-all works sufficiently well. What would be the drawback of using this catch-all wording?

Reviewer 2: @carmenbianca, well, let’s say you’re feeling pretty smart and smug and store code that you don’t want to have tools to find, but still plant in let’s say something you hold patents in.

How would you prevent such a malicious actor to

- simply store the patent-infested code into the `.reuse/` folder,
- not attach and FOSS license to that code
- in the rest of the code call code files from that folder, so that it pulls it in when building or running
- wait and laugh while the others make use of the trojan package

The package/repo will be REUSE compliant, according to REUSE Spec it will look FOSS as well. But as soon as  someone uses it or implements it into their own product/project, the malicious actor may strike with a patent lawsuit.

Author: @mxmehl Opinion?

@silverhook One way or another there is going to have to be a catch-all here, because the tool will (in the future) allow the user to put custom templates in there.

It could be half-fixed by saying something like "the configuration and template files in `.reuse/`", but this might force an update of the spec in case the tool ever decides to implement new stuff.

Reviewer 2: You can always specify how the templates should look like and how their files have to be named.

<!-- TODO: The files inside of a git submodule -->

There are various ways to associate copyright and licensing information with a
file.

### Comment headers

> Each plain text file that can contain comments should contain comments at the

Reviewer 2: ```suggestion
To implement this method each plain text file that can contain comments MUST contain comments at the
```

Author: LGTM

Author: Fixed in c17e4d43b3a6b74ffa5de3004decae3348fba2c3

top of the file (comment header) that declare that file's copyright and
licensing information.

If a file is not a plain text file or does not permit the inclusion of
comments, the comment header that declares the file's copyright and licensing
information should be in an adjacent file of the same name with the
additional extension `.license` (example: `cat.jpg.license` if the original
file is `cat.jpg`).

> The comment header must contain one or more `SPDX-Copyright` tags, and one or

Reviewer 2: ```suggestion
The comment header MUST contain one or more `SPDX-Copyright` tags, and one or
```

Reviewer 3: FYI - The SPDX-Copyright tag is not yet included in the spec.  We are tracking an issue to add it here: https://github.com/spdx/spdx-spec/issues/122

Author: Correct, I'm aware. This spec won't be published until there is a consensus in SPDX about such an added tag. Thanks for the link to the issue, though!

Author: Fixed in 54b7d330d56ce1912f6d949f8745730aa20b1fc5, though still waiting on https://github.com/spdx/spdx-spec/issues/122

more `SPDX-License-Information` tags. A tag is followed by a colon, followed by
a text value, and terminated by a newline.

The `SPDX-Copyright` tag must be followed by a copyright notice as described
in <!-- FIXME: Refer to an internal section -->.

> Instead of the `SPDX-Copyright` tag, the symbols `Copyright`, `Copyright (c)`,

Reviewer 2: ```suggestion
```
I suggest we make the `SPDX-Copyright` tag in REUSE 3.0 mandatory.

The main goal of REUSE is to make licensing and copyright info human and machine readable.

People will need to edit their headers anyway, so prepending a tag to their copyright notices should be negligibe extra work. On the other hand, having a simple tag to search for copyright notices simplifies the search immensely.

Reviewer: @kestewart We discussed this yesterday, and you seemed to be relieved that SPDX-Copyright is not mandatory, even after the release of SPDX 2.2. Did I understand you correctly, or would you be fine with making this mandatory for REUSE 3.0?

Reviewer: AFAIK, Thomas does not use SPDX-Copyright for his current overhaul of the licensing information in Linux. I wouldn't like to break his efforts...

Author: I wouldn't make SPDX-Copyright mandatory, just to stay compatible with all the ways that people have been doing copyright notices thus far. I would strongly recommend it, though.

@mxmehl Can you communicate with Thomas to see if he would use SPDX-Copyright? Though it's a WIP at https://github.com/spdx/spdx-spec/issues/122

Reviewer: Good idea, will send a mail to the linux-spdx list today

> `Copyright (C)`, `Copyright ©` or `©` may be used.

Reviewer 2: ```suggestion
```

Author: Closing this in favour of a different convo

> The `SPDX-License-Identifier` tag must be followed by a valid SPDX License

Reviewer 2: ```suggestion
The `SPDX-License-Identifier` tag MUST be followed by a valid SPDX License
```

Expression describing the licensing of the file (example:
`SPDX-License-Identifier: GPL-3.0-or-later OR Apache-2.0`). If separate sections
of the file are licensed differently, a different `SPDX-License-Identifier` tag
should be included for each section.

An example of a comment header:

```
# SPDX-Copyright: 2016, 2018-2019 Jane Doe <jane@example.com>
# SPDX-Copyright: 2019 Example Company
#
# SPDX-License-Identifier: GPL-3.0-or-later
```

### DEP5

> Copyright and licensing information may be associated with a file through a DEP5

Reviewer 2: ```suggestion
Alternatively, copyright and licensing information MAY be associated with a file through a DEP5
```

Author: Fixed in c27e040e3f803f3f775a2c81a57a9ef1a4f9bea8

> file. The intended use case of this method is large directories where including

Reviewer 2: ```suggestion
File. The intended use case of this method is large directories where including 
```

a comment header in each file (or in `.license` companion files) is impossible
or undesirable.

> The DEP5 file must be in `.reuse/dep5` from the root of the project.

Reviewer 2: ```suggestion
The DEP5 File MUST be named `dep5` and stored in the `.reuse/` directory in from the root of the project (i.e. `.reuse/dep5`).
```

Author: project -> Project

Author: Fixed in 450c0c7cdfd722ce4b1873d09a3a070c69c4ca02

> <!-- TODO: How to do multiple licenses without AND and OR. -->

Reviewer 2: DEP5 does make use both of `and` and `or` as well as brackets `(` and `)` for grouping – identical to SPDX syntax.

There are some other incompatibilities between DEP5 and SPDX though, mainly in the names of some of the licenses:
https://wiki.debian.org/Proposals/CopyrightFormat#Differences_between_DEP5_and_SPDX

Author: I know, but I'm trying to do the following:

```
SPDX-License-Identifier: MIT
SPDX-License-Identifier: 0BSD
```

This is different from `MIT AND 0BSD` and `MIT OR 0BSD`. It means that part of the file is under MIT, and part is under 0BSD.

`MIT AND 0BSD` suggests that both licenses apply to the contents of the file and that they must both be obeyed, which is really really close.

`MIT OR 0BSD` suggests that the user may choose which license applies to them, which is not correct.

Author: Oh, as addendum, another way that the SPDX spec deals with this is:

```
SPDX-License-Identifier: MIT; 0BSD
```

(but the tool currently does not support this :x)

Reviewer 3: Could you just list separate licenses on separate SPDX-License-Identifier lines as described above?  In converting to an SPDX document, the tools which parse the source files would need to convert them to an expression for the file declared license, but it could also add them individually to the `PackageLicenseInfoFromFiles` which is a list of licenses collected from the source files.

Author: That's what we normally do in source code files, but that isn't supported inside of the DEP5 spec, that I know of.

It's a bit of a pedantic thing, though, because the operators `AND` and `;` are _almost_ identical.

Author: Coming back to this: I suggested that the SPDX spec allows an expression such as

```
SPDX-License-Identifier: MIT; 0BSD
```

This is false. I researched this further today, and could not find it in the spec. Instead, I had previously found it [here](https://wiki.spdx.org/view/FileNoticeExamples#Package_Examples), but had not read the disclaimer that the spec does not (yet?) support this.

Reviewer 3: I can confirm that most license expression parsers don't support this operator.

In any case, I would recommend not using the `;` as most SPDX license expression parsers will fail and there is no agreed to definition in the SPDX spec.

I added an issue to the SPDX Spec to track either adding to to the spec or removing it from the wiki page to avoid future confusion: https://github.com/spdx/spdx-spec/issues/123

The `License` tag must be followed by a valid SPDX License Expression describing
the licensing of the associated files.

The `Copyright` tag must be followed by a copyright notice as described in <!--
FIXME: Refer to an internal section -->.

An example of a DEP5 file:

```
Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
Upstream-Name: Project
Upstream-Contact: Jane Doe <jane@example.com>
Source: https://example.com/jane/project

Files: po/*
Copyright: 2019 Translation Company
License: GPL-3.0-or-later
```

## Format of copyright notices

> A copyright notice must start with the copyright symbol (©), the word

Reviewer 2: ```suggestion
A copyright notice MUST start with the copyright symbol "©", or the word
```

> "Copyright", or an equivalent tag, followed by the year(s) of publication, the

Reviewer 2: ```suggestion
word "Copyright", followed by the year of first publication (a good example would be the year that specific file was created), the
```
I would also remove the `or the word "Copyright"`, as apart from in the US (and other English-speaking jurisdictions), strictly speaking, it is not a valid copyright notice.

See also https://github.com/fsfe/reuse-docs/issues/5

Author: I agree with your reasoning, but it's there mostly for backwards compatibility. Let's continue this discussion in the other thread where I mentioned the Linux kernel.

Author: Oh, I think there is a slight misunderstanding here. When I wrote "or an equivalent tag", I meant "SPDX-Copyright: " (SPDX) and "Copyright: " (DEP5).

Reviewer 2: I see where the misunderstanding came from. In any case we should reword it in order to avoid such misunderstanding.

Author: Fixed in 9956e990765be298524249146a0eb9140ecf7590

> name of the copyright holder, and the e-mail address of the copyright holder.

Reviewer 2: ```suggestion
name of the copyright holder, and a contact of the copyright holder. In case the copyright holder is a natural person, their e-mail address would be a good contact; in case the copyright holder is a legal entity, its website would be a good point of contact.
```

Author: Agree with the change from e-mail address to contact. I would rephrase the second sentence to be shorter, however, and include the word "example".

Reviewer 2: Sounds like a good change to me.

Author: Fixed in 9956e990765be298524249146a0eb9140ecf7590

> The order of these items may be changed, but should be consistent within the

Reviewer 2: ```suggestion
```

What purpose would that serve?

The only reason I can see is that it would make the plethora of existing malformed copyright notices REUSE compliant.

If that is the desire, I would at least hope RUSE has a strong recommendation for how a proper copyright notice should look like.

Author: Slight agree. @mxmehl ?

Reviewer: Hm, I have also seen a few other forms than what we recommend here. IANAL, but I don't see a big reason for making this format a MUST. REUSE aims at making software reusable, not pretty licensing information ;)

So I would prefer a strong recommendation but allow other forms in the linter

Author: I'm leaning towards agreeing with @mhmehl.

Reviewer 2: I have seen all sorts of things put as copyright statements in the wild.

As discussed before in #5 there is a typical way how copyright statements are described by law(s).

In any case, different from a license, the copyright statement does not impact directly the rights granted, or even who the holder/granter of these rights is. It is (just) an indication who (apparently) holds the rights though. As such, _legally_ it won’t make a difference whether we put some strict rules how this should be formatted or keep it completely free text. The difference is just how easy it will be to parse it.

Author: Maybe possibly fixed in 9956e990765be298524249146a0eb9140ecf7590?

Reviewer 2: I’m OK with the state as it is in https://github.com/fsfe/reuse-docs/pull/23/commits/5bbf9573d6ecde014dcc55786a3b7710775c26a3 , but will submit some proposed changes of wording, to make it even more unambiguous.

Author: Then closing this convo

> project. The items except the name of the copyright holder may be omitted.

Reviewer 2: ```suggestion
```
Same as above.

Author: Disagree on this front. Microsoft does not include the year in their copyright notices. They have their reasons for that, and I wouldn't like to see REUSE be incompatible with their interpretation of copyright law.

Reviewer: Slight agreement

Reviewer 2: > Slight agreement

With whom? :)

Author: Closing this without issue because this section is being reworked anyway.

> Examples of valid copyright notices:

Reviewer 2: Again, ideally, I would like there to be only on valid copyright notice:

```
SPDX-Copyright: © 2019 Acorn Hacker <a.hacker@example.org>
SPDX-Copyright: © 2011 Some Company <https://www.somecompany.example>
```

Author: Big disagree.

This is for backwards compatibility. We really want REUSE to be compatible with what the Linux kernel is currently doing. The `SPDX-Copyright` tag is recommended with a SHOULD, and is exclusively mentioned in the tutorial (and I believe FAQ). The rest is just backwards compatibility.

Reviewer 2: I have a very strong opinion on this, but won’t fight tooth and nail for it, as long as it also allows for the following to be REUSE valid (with or without an additional SPDX tag before it):

```
© 2019 Acorn Hacker <a.hacker@example.org>
© 2019 Muh Company <https://muh.example>
```l

Ultimately, the copyright field could be a free text in the Spec, and to have a (strong) sane recommendation in the Tutorial (and FAQ).

We can always make the 4.0 or 16.0 version of the Spec stricter in that regard, once the suggestions in the Tutorial and FAQ spread out more.

followed by a `License-Text` tag, followed by the license text. The value
following a `Valid-License-Identifier` tag MUST be a valid SPDX Identifier of
the license text contained within the License File.

Example:

```
Valid-License-Identifier: GPL-3.0-only
Valid-License-Identifier: GPL-3.0-or-later
License-Text:

                    GNU GENERAL PUBLIC LICENSE
                       Version 3, 29 June 2007

 Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
 Everyone is permitted to copy and distribute verbatim copies
 of this license document, but changing it is not allowed.

[...]
```

A Project MAY contain a single License File outside of the `LICENSES/` directory
in the file `COPYING`, `COPYRIGHT`, `LICENSE` or `LICENCE` in the root of the
Project.

If a license does not exist in the SPDX License List, its SPDX Identifier
MUST be `LicenseRef-[idstring]` as defined by the SPDX Specification.

A Project MUST NOT include License Files for licenses under which none of the
files in the Project are licensed.

Everything that applies to licenses in this section also applies to license
exceptions, with the following specifics:

> 

Author: I think this is better covered by the SPDX spec.

> 

Author: Closing this without issue because we're in agreement.

- Exception files MUST be placed in the `LICENSES/` directory in the root of the
  Project.

- The tags `Valid-Exception-Identifier` and `Exception-Text` MUST be used
  instead of `Valid-License-Identifier` and `License-Text`.

- It is not possible to have a license exception that does not exist in the SPDX

>   License List.

Author: This seems a little too wordy to me, but I can +1 the intention.

  `.git/`).

- The files ignored by the version control system (example: Files listed in
  `.gitignore`).

- The files in the `.reuse/` directory in the root of the Project.

<!-- TODO: The files inside of a git submodule -->

There are two ways to associate Copyright and Licensing Information with a
file.

### Comment headers

To implement this method, each plain text file that can contain comments MUST
contain comments at the top of the file (comment header) that declare that
file's Copyright and Licensing Information.

If a file is not a plain text file or does not permit the inclusion of
comments, the comment header that declares the file's copyright and licensing
information SHOULD be in an adjacent file of the same name with the
additional extension `.license` (example: `cat.jpg.license` if the original
file is `cat.jpg`).

The comment header MUST contain one or more `SPDX-Copyright` tags, and one or
more `SPDX-License-Information` tags. A tag is followed by a colon, followed by
a text value, and terminated by a newline.

The `SPDX-Copyright` tag MUST be followed by a copyright notice as described
in <!-- FIXME: Refer to an internal section -->.

> Instead of the `SPDX-Copyright` tag, the word `Copyright` or the symbol `©` MAY

Author: Swapped around in 5bbf9573d6ecde014dcc55786a3b7710775c26a3

be used.

The `SPDX-License-Identifier` tag MUST be followed by a valid SPDX License
Expression describing the licensing of the file (example:

> `SPDX-License-Identifier: GPL-3.0-or-later OR Apache-2.0`). If separate sections

Author: Closing this without issue because we're in agreement on the previous, similar thread. Please open an issue if I closed this erroneously.

of the file are licensed differently, a different `SPDX-License-Identifier` tag
MUST be included for each section.

An example of a comment header:

```
# SPDX-Copyright: 2016, 2018-2019 Jane Doe <jane@example.com>
# SPDX-Copyright: 2019 Example Company
#
# SPDX-License-Identifier: GPL-3.0-or-later
```

### DEP5

Alternatively, Copyright and Licensing Information MAY be associated with a
file through a DEP5 file. The intended use case of this method is large
directories where including a comment header in each file (or in `.license`
companion files) is impossible or undesirable.

> DEP5 File MUST be named `dep5` and stored in the `.reuse/` directory in the root

Author: There's still a conversation open about that. I'll add a word "the" to that sentence, though, because it badly needs one.

root of the Project (i.e. `.reuse/dep5`).

<!-- TODO: How to do multiple licenses without AND and OR. -->

The `License` tag MUST be followed by a valid SPDX License Expression describing
the licensing of the associated files.

The `Copyright` tag MUST be followed by a copyright notice as described in <!--
FIXME: Refer to an internal section -->.

An example of a DEP5 file:

```
Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
Upstream-Name: Project
Upstream-Contact: Jane Doe <jane@example.com>
Source: https://example.com/jane/project

Files: po/*
Copyright: 2019 Translation Company
License: GPL-3.0-or-later
```

## Format of copyright notices

A copyright notice MUST be prefixed by a tag, symbol or word denoting a
copyright notice as described in this specification.

> The copyright notice MUST contain the year(s) of publication, the name of the

Reviewer 2: ```suggestion
The copyright notice SHOULD contain the year(s) of publication, the name of the
```

> copyright holder, and the contact address of the copyright holder. The order of

Reviewer 2: ```suggestion
copyright holder(s), and the contact address of the copyright holder(s); whereby the name of the copyright holder(s) MUST be included. The order of
```

Author: Multiple copyright holders in a single copyright notice?

Reviewer 2: Seen that happen before.

But more sane, you have unincorporated groups of copyright holders such as:

```
© 2001 Contributors to Project X (see file `AUTHORS`)
```

```
Copyright 1998-2019 Project Y
```

I’m fine with omitting the “(s)” in “holder(s)” though, if anyone feels strongly about that.

> these items MAY be changed, but SHOULD be consistent within the Project. The

Reviewer 2: ```suggestion
these items MAY be changed, but SHOULD be consistent within the Project.
```

> items except the name of the copyright holder MAY be omitted.

Reviewer 2: ```suggestion
```
Before we were running into a “MUST vs SHALL” problem on the same piece of info. I think my suggestion makes it less ambiguous, but I’m very open to better and nicer wording.

The `SPDX-Copyright` tag MUST be followed by a copyright notice as described
in <!-- FIXME: Refer to an internal section -->.

Instead of the `SPDX-Copyright` tag, the symbol `©`, or the word `Copyright` MAY

> be used, in which case a colon is not needed..

Reviewer 2: Unneeded extra dot at the end of line 144, but otherwise looks great and thanks, @carmenbianca  :)

Author: Oops, this is what I get for copy-pasting without verifying.

Will fix, thanks
